File: Chapter02/example_topic_declare.rb
#	1. Require client library
require "bunny"

#	2. Read RABBITMQ_URI from ENV
connection = Bunny.new ENV["RABBITMQ_URI"]

#	3. Start a communication session with RabbitMQ
connection.start
channel = connection.create_channel

def on_start(channel)
  #4. Declare and return the topic exchange, taxi-topic
  channel.topic("taxi-topic", durable: true, auto_delete: true)
end
exchange = on_start(channel)

File: csharp_code/Chapter02/example_topic_declare.cs
using System;
using System.Collections.Generic;
using System.Text;
using RabbitMQ.Client;
using RabbitMQ.Samples.Common;

namespace RabbitMQ.Samples.Chapter02
{
    public class ExampleTopicDeclare
    {
        public static void DeclareTopic(RabbitMqConnection rabbitMqConnection)
        {
            var channel = rabbitMqConnection.Channel;

            // Declare a direct exchange, taxi-direct
            channel.ExchangeDeclare(Chapter2Constants.TopicExchange,
                ExchangeType.Topic,
                durable: true,
                autoDelete: true);
        }
    }
}
